tests/src/test/java/io/streamnative/pulsar/handlers/kop/SchemaRestApiTest.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.resources.SubjectResource;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import lombok.Cleanup;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test the Schema related REST APIs.
+ */
+public class SchemaRestApiTest extends KopProtocolHandlerTestBase {
+
+    protected static final ObjectMapper MAPPER = new ObjectMapper()
+            .configure(SerializationFeature.INDENT_OUTPUT, true);
+    private String baseUrl;
+
+    @BeforeClass
+    @Override
+    protected void setup() throws Exception {
+        super.enableSchemaRegistry = true;
+        this.internalSetup();
+        baseUrl = "http://localhost:" + conf.getKopSchemaRegistryPort();
+    }
+
+    @AfterClass
+    @Override
+    protected void cleanup() throws Exception {
+        this.internalCleanup();
+    }
+
+    @Test
+    public void testDeleteSubject() throws Exception {
+        final var createSchemaRequest = new SubjectResource.CreateSchemaRequest();
+        createSchemaRequest.setSchema("{\"type\":\"record\",\"name\":\"User1\",\"namespace\":\"example.avro\""
+                + ",\"fields\":[{\"name\":\"name\",\"type\":\"string\"},{\"name\":\"age\",\"type\":\"int\"}]}");
+        final var subject = "my-subject";
+        sendHttpRequest("POST", "/subjects/" + subject + "/versions",
+                MAPPER.writeValueAsString(createSchemaRequest));
+
+        assertEquals(getSubjects(), Collections.singletonList(subject));
+        resetSchemaStorage();
+        assertEquals(getSubjects(), Collections.singletonList(subject));
+
+        sendHttpRequest("DELETE", "/subjects/" + subject, null);
+
```

Comment:
```suggestion
        assertTrue(getSubjects().isEmpty());
```
We can assert first before resetting schema storage.